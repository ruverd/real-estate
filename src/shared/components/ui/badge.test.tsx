import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { Badge } from "./badge";

const renderBadge = (props?: React.ComponentProps<typeof Badge>) => {
  return render(<Badge {...props}>Test Badge</Badge>);
};

describe("Badge", () => {
  it("should render with default variant", () => {
    renderBadge();

    const badge = screen.getByText("Test Badge");

    expect(badge).toBeInTheDocument();
  });

  it("should render with outline variant", () => {
    renderBadge({ variant: "outline" });

    const badge = screen.getByText("Test Badge");

    expect(badge).toBeInTheDocument();
  });

  it("should apply custom className", () => {
    const customClass = "custom-class";

    renderBadge({ className: customClass });

    const badge = screen.getByText("Test Badge");

    expect(badge).toHaveClass(customClass);
  });

  it("should match snapshot with default variant", () => {
    const { container } = renderBadge();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should match snapshot with outline variant", () => {
    const { container } = renderBadge({ variant: "outline" });

    expect(container.firstChild).toMatchSnapshot();
  });
});
