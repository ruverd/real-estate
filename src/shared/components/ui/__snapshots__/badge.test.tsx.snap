// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > should match snapshot with default variant 1`] = `
<span
  class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90"
  data-slot="badge"
>
  Test Badge
</span>
`;

exports[`Badge > should match snapshot with outline variant 1`] = `
<span
  class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden text-foreground [a&]:hover:bg-accent [a&]:hover:text-accent-foreground"
  data-slot="badge"
>
  Test Badge
</span>
`;
